.github/workflows/build.yml

```
@@ -2,7 +2,10 @@
 name: 'Build'
 
 on:
+  pull_request:
   push:
+    branches:
+      - main
```

Isn't this too restrictive? Seems to be that this workflow is the most useful one to be run on every commit, and it is reasonably fast.
And also, if we are restricting on branches, it might be good to add a predefined branch for tests/development like you did on the debian one? Something like:
```suggestion
      - main
      - workflows/*
```